src/main/java/dao/H2Factory/H2CustomerDao.java

```diff
@@ -2,108 +2,72 @@
+import org.springframework.stereotype.Repository;
 
+@Repository("h2CustomerDao")
 public class H2CustomerDao implements CustomerDaoInterface {
 
-    private final ConnectionProvider connectionProvider;
+    @Autowired
+    private NamedParameterJdbcTemplate jdbcTemplate;
     private static final Logger LOG = LoggerFactory
             .getLogger(H2CustomerDao.class);
-    private Converting<Customer> converter;
 
-    protected H2CustomerDao(ConnectionProvider connectionProvider) {
-        this.connectionProvider = connectionProvider;
-        converter = CustomerConverter.getInstance();
+    H2CustomerDao() {
     }
 
-    public void create(Customer newCustomer) throws DaoException {
-        try {
-            PreparedStatement st = connectionProvider.getConnection()
-                    .prepareStatement("insert into customerapplication"
-                            + ".customer(firstname,lastname,"
-                            + "address) values(?,?,?)");
-            st.setString(1, newCustomer.getFirstName());
-            st.setString(2, newCustomer.getName());
-            st.setString(3, newCustomer.getAddress());
-            st.executeUpdate();
-            connectionProvider.destroy();
-        } catch (SQLException ex) {
-            LOG.error(ex.getSQLState());
-            throw new DaoException("Inserting data error", ex);
-        }
+    public void create(Customer newCustomer) {
+        String query = "insert into customerapplication"
+                + ".customer(firstname,lastname,"
+                + "address) values(:firstname,:lastname,:address)";
+        Map namedParameters = new HashMap();
+        namedParameters.put("firstname", newCustomer.getFirstName());
+        namedParameters.put("lastname", newCustomer.getName());
+        namedParameters.put("address", newCustomer.getAddress());
+        jdbcTemplate.update(query, namedParameters);
     }
 
-    public Customer get(int id) throws DaoException {
-        Customer gotCustomer = null;
-        try {
-            PreparedStatement st = connectionProvider.getConnection()
-                    .prepareStatement("select * from customerapplication"
-                            + ".customer where id=?");
-            st.setInt(1, id);
-            ResultSet rs = st.executeQuery();
-            gotCustomer = converter.convert(rs).get(0);
-            connectionProvider.destroy();
-        } catch (SQLException ex) {
-            LOG.error(ex.getSQLState());
-            throw new DaoException("Selection data error", ex);
-        }
+    public Customer get(int id) {
+        String query = "select * from customerapplication"
+                + ".customer where id=:customerId";
+        Map namedParameters = new HashMap();
+        namedParameters.put("customerId", id);
+        Customer gotCustomer = (Customer) jdbcTemplate
+                .queryForObject(query, namedParameters, new CustomerMapper());
```

Comment:
Customer Mapper also can be injected through spring - code will be more testable.